docs/source/api/user-guide/measurements/randomized-response.rst

```diff
@@ -0,0 +1,323 @@
+            estimated: ...
+
+As expected, the bias correction admits a useful estimate of the
+population proportion (``.23``).
+
+The categorical randomized response will suffer the same bias:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> import numpy as np
+            >>> num_responses = 1000
+            
+            >>> true_probability = [0.1, 0.4, 0.3, 0.2]
+            
+            >>> private_cat_responses = []
+            
+            >>> for _ in range(num_responses):
+            ...     response = np.random.choice(categories, p=true_probability)
+            ...     randomized_response = rr_meas(response)
+            ...     private_cat_responses.append(randomized_response)
+            
+            >>> from collections import Counter
+            
+            >>> counter = Counter(private_cat_responses)
+            >>> naive_cat_proportions = [counter[cat] / num_responses for cat in categories]
+            >>> naive_cat_proportions
+            [..., ..., ..., ...]
+
+We can do the same analysis to de-bias the categorical estimate:
+
+.. dropdown:: Derivation of Categorical RR Bias Correction
+
+   Denote the randomized response :math:`Y_i = \texttt{rr\_meas}(X_i)`, and
+   the :math:`k^{th}` category as :math:`C_k`.
+
+   We first find :math:`E[I[Y_i = C_k]]` (the expectation that noisy
+   responses are equal to the :math:`k^{th}` category). This is done by
+   considering the law of total probability over all categories.
+
+   .. math::
+
+      \begin{align*}
+         E[I[Y_i = C_k]] &= p \cdot I[X_i = C_k] + \sum_{j \ne k} \frac{1 - p}{K - 1} \cdot I[X_i = C_j] \\
+            &= p \cdot I[X_i = C_k] + \frac{1 - p}{K - 1} \cdot (1 - I[X_i = C_k])
+      \end{align*}
+
+   Then solve for :math:`E[I[X_i = C_k]]` (the expectation that raw
+   responses are equal to the :math:`k^{th}` category):
+
+   .. math::
+
+      \begin{align*}
+         E[I[Y_i = C_k]] (K - 1) &= p \cdot E[I[X_i = C_k]] (K - 1) + (1 - p)(1 - E[I[X_i = C_k]]) \\
+         E[I[Y_i = C_k]] (K - 1) &= p \cdot E[I[X_i = C_k]] K - p - E[I[X_i = C_k]] + 1 \\
+         E[I[Y_i = C_k]] (K - 1) + p - 1 &= E[I[X_i = C_k]] (pK - 1) \\
+         \frac{E[I[Y_i = C_k]] (K - 1) + p - 1}{pK - 1} &= E[I[X_i = C_k]]
+      \end{align*}
+
+
+This formula is represented in the following function:
+
+.. tab-set::
+
+    .. tab-item:: Python
+        :sync: python
+
+        .. code:: python
+
+            >>> def debias_randomized_response(mean_releases, p):
+            ...     """Adjust for the bias of the mean of a categorical RR dataset."""
+            ...     mean_releases = np.array(mean_releases)
+            ...     assert all(mean_releases >= 0) and abs(sum(mean_releases) - 1) < 1e-6
+            ...     assert 0 <= p <= 1
+            ...     
+            ...     k = len(mean_releases)
+            ...     return (mean_releases * (k - 1) + p - 1) / (p * k - 1)
```

Comment:
Similarly here, this should be a library utility. Future issue.

Reply:
- Filed #1806